exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/_mapping.py

```diff
@@ -0,0 +1,135 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from re import UNICODE, compile
+
+_SANITIZE_NAME_RE = compile(r"([^a-zA-Z0-9:]+)|_{2,}", UNICODE)
+# Same as name, but doesn't allow ":"
+_SANITIZE_ATTRIBUTE_KEY_RE = compile(r"([^a-zA-Z0-9]+)|_{2,}", UNICODE)
+
+# UCUM annotations are ASCII chars 33-126 enclosed in curly braces
+# https://ucum.org/ucum#para-6
+_UCUM_ANNOTATION_CURLY = compile(r"{[!-~]*}")
+
+# Remaps common UCUM and SI units to prometheus conventions. Copied from
+# https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.101.0/pkg/translator/prometheus/normalize_name.go#L19
+# See specification:
+# https://github.com/open-telemetry/opentelemetry-specification/blob/v1.33.0/specification/compatibility/prometheus_and_openmetrics.md#metric-metadata-1
+_UNIT_MAPPINGS = {
+    # Time
+    "d": "days",
+    "h": "hours",
+    "min": "minutes",
+    "s": "seconds",
+    "ms": "milliseconds",
+    "us": "microseconds",
+    "ns": "nanoseconds",
+    # Bytes
+    "By": "bytes",
+    "KiBy": "kibibytes",
+    "MiBy": "mebibytes",
+    "GiBy": "gibibytes",
+    "TiBy": "tibibytes",
+    "KBy": "kilobytes",
+    "MBy": "megabytes",
+    "GBy": "gigabytes",
+    "TBy": "terabytes",
+    # SI
+    "m": "meters",
+    "V": "volts",
+    "A": "amperes",
+    "J": "joules",
+    "W": "watts",
+    "g": "grams",
+    # Misc
+    "Cel": "celsius",
```

Comment:
```suggestion
    "Cel": "degree celsius",
```

From [this](https://www.bipm.org/documents/20126/41483022/SI-Brochure-9-concise-EN.pdf/2fda4656-e236-0fcb-3867-36ca74eea4e3).